perun/postprocess/regression_analysis/methods.py

```diff
@@ -0,0 +1,445 @@
+    computed. Only the currently best fitting model is computed in each step.
+
+    This method might produce only local result (local extrema), but it's generally faster than
+    the full computation method.
+
+    Arguments:
+        x_pts(list): the list of x points coordinates
+        y_pts(list): the list of y points coordinates
+        computation_models(tuple of str): the collection of regression models to compute
+        kwargs: additional configuration parameters
+    Raises:
+        GenericRegressionExceptionBase: derived versions which are used in the computation functions
+        DictionaryKeysValidationFailed: in case the data format dictionary is incorrect
+    Returns:
+        iterable: the generator object which produces best fitting model as a transformed data
+                  dictionary
+
+    """
+    x_pts, y_pts = tools.shuffle_points(x_pts, y_pts)
+
+    # Do the initial step for specified models
+    model_generators, results = _models_initial_step(x_pts, y_pts, computation_models,
+                                                     kwargs['steps'])
+    best_fit = -1
+    while True:
+        try:
+            # Get the best fitting model and do next computation step
+            best_fit = _find_best_fitting_model(results)
+            results[best_fit] = next(model_generators[best_fit])
+        except StopIteration:
+            # The best fitting model finished the computation, end of computation
+            yield results[best_fit]
+            break
+
+
+def interval_computation(x_pts, y_pts, computation_models, **kwargs):
+    """The interval computation method.
+
+    This method splits the regression data into evenly distributed sorted parts (i.e. intervals)
+    and each interval is computed separately using the full computation.
+
+    This technique allows to find different regression models in each interval and thus discover
+    different complexity behaviour for algorithm based on it's input size.
+
+    Arguments:
+        x_pts(list): the list of x points coordinates
+        y_pts(list): the list of y points coordinates
+        computation_models(tuple of str): the collection of regression models to compute
+        kwargs: additional configuration parameters, 'steps' expected
+    Raises:
+        GenericRegressionExceptionBase: derived versions which are used in the computation functions
+        DictionaryKeysValidationFailed: in case the data format dictionary is incorrect
+    Returns:
+        the generator object which produces computed models one by one for every interval as
+        a transformed output data dictionary
+
+    """
+    # Sort the regression data
+    x_pts, y_pts = tools.sort_points(x_pts, y_pts)
+    # Split the data into intervals and do a full computation on each one of them
+    for part_start, part_end in tools.split_sequence(len(x_pts), kwargs['steps']):
+        interval_gen = full_computation(x_pts[part_start:part_end], y_pts[part_start:part_end],
```

Comment:
Does this need to be full_computation? The functions should have the same API, right? So could full_computation be a parameter to this function? The easiest would be to make full_computation as default parameter.